pkgs/top-level/platforms.nix

```diff
@@ -443,4 +443,12 @@ rec {
     };
   };
 
+  chooseDefault = { system }:
+    if system == "armv6l-linux" then raspberrypi
```

Comment:
nit: rename `chooseDefault` by `selectPlatformBySystem`.  No need to wrap the argument in an attribute set.